megatron/core/inference/contexts/dynamic_context.py

```diff
@@ -946,94 +926,118 @@ def reset_mamba_state(self) -> None:
             self.mamba_ssm_states.fill_(0)
             self.mamba_metadata.reset()
 
-    def using_cuda_graph_this_step(self) -> bool:
-        """Returns True if cuda graphs are being used for this step."""
-        has_cuda_graphs = self.cuda_graph_token_counts is not None
-        can_use_cuda_graphs = self.is_decode_only() or self.non_decode_cuda_graphs
-        token_count_fits_cuda_graph = has_cuda_graphs and (
-            self.active_token_count <= self.max_cuda_graph_token_count
+    def add_dummy_requests_for_cudagraph_capture(
+        self, graph_dimensions: InferenceBatchDimensions
+    ) -> None:
+        """
+        Adds dummy requests to reflect the number of prefill and decode requests in the graph config.
+        These are using during cuda graph captures.
+        """
+        prefill_tokens = graph_dimensions.token_count - graph_dimensions.decode_req_count
+
+        for i in range(graph_dimensions.decode_req_count):
+            self.add_request(
+                DynamicInferenceRequest(
+                    request_id=i,
+                    prompt_tokens=torch.zeros(
+                        1, dtype=torch.long, device=torch.cuda.current_device()
+                    ),
+                    sampling_params=SamplingParams(num_tokens_to_generate=1, termination_id=-1),
+                )
+            )
```

Comment:
Did we land on using a for loop for this as opposed to directly constructing the metadata tensors? Do we have profile data for this loop?